Stratis.Bitcoin/Features/BlockStore/BlockStoreLoopStep.cs

```diff
@@ -78,41 +73,11 @@ protected BlockStoreLoopStep(BlockStoreLoop blockStoreLoop)
     ///     <item>3: "Next" > Execute the next line of code.</item>
     /// </list>
     /// </para>
-    /// </summary>
-    public sealed class BlockStoreLoopStepResult
+    /// </summary>   
+    public enum StepResult
     {
-        private BlockStoreLoopStepResult() { }
-
-        public bool ShouldBreak { get; private set; }
-        public bool ShouldContinue { get; private set; }
-
-        /// <summary>
-        /// Will cause the caller to break out of the iteration.
-        /// </summary>
-        internal static BlockStoreLoopStepResult Break()
-        {
-            var result = new BlockStoreLoopStepResult();
-            result.ShouldBreak = true;
-            return result;
-        }
-
-        /// <summary>
-        /// Will cause the caller to Continue execution of the loop i.e. go onto the next item in the iteration.
-        /// </summary>
-        internal static BlockStoreLoopStepResult Continue()
-        {
-            var result = new BlockStoreLoopStepResult();
-            result.ShouldContinue = true;
-            return result;
-        }
-
-        /// <summary>
-        /// Will cause the caller to execute the next line of code.
-        /// </summary>
-        internal static BlockStoreLoopStepResult Next()
-        {
-            var result = new BlockStoreLoopStepResult();
-            return result;
-        }
+        Continue,
+        Next,
+        Stop,
```

Comment:
Move comment from enum to each of the values. We always want the enum values to be commented - you have those comments already, but in wrong place.